nui/src/main/java/org/terasology/nui/util/NUIMathUtil.java

```diff
@@ -33,15 +33,23 @@ private NUIMathUtil() {
     /**
      * a + b, but if the result exceeds Integer.MAX_VALUE then the result will be Integer.MAX_VALUE rather than overflowing.
      *
-     * @param a
-     * @param b
      * @return min(a + b, Integer.MAX_VALUE)
      */
     public static int addClampAtMax(int a, int b) {
         long result = (long) a + (long) b;
         return (int) Math.min(result, Integer.MAX_VALUE);
     }
 
+    /**
+     * a + b, but if the result exceeds Float.MAX_VALUE then the result will be Float.MAX_VALUE rather than overflowing.
+     *∂
+     * @return min(a + b, Float.MAX_VALUE)
+     */
+    public static float addClampAtMax(float a, float b) {
+        double result = (double) a + (double) b;
+        return (float) Math.min(result, Float.MAX_VALUE);
+    }
+
```

Comment:
There is a clamp utility under `org.joml.Math`

Reply:
I don't think this will help here: JOML's clamp is the usual `max(a, min(b, val))` - but what we need is an upcast (int >> long) for the addition and then check for an overflow.

Reply:
Oh, I got a bit confused. yea this should be fine.